### PR TITLE
Finishes PR #600

### PR DESCRIPTION
**Pull request by dcalvoalonso**
- Rebases branch and solves conflicts.
- Refactors PR code to leverage on existing functionality introduced in #664.

All unit tests (also the ones added in this PR) are passed.

Linting is checked.

### CODE REVIEW

**Comment by dcalvoalonso** [comment]
I have also added some unit tests for NGSIv2 since the functionality is already implemented.

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -2,4 +2,7 @@ Fix: modified engine version >=6 in packages.json
 Fix: update logops dependency from 1.0.8 to 2.1.0
 Fix: user provider timestamp correctly propagated to mapped entities when multientity pluging is used (#748)
 Add: new functions to set callbacks for removal of devices and groups (#735)
-Fix: add ?type parameter in CB request updates to avoid potential entity ambiguities (#733)
\ No newline at end of file
+Fix: add ?type parameter in CB request updates to avoid potential entity ambiguities (#733)
+Add: support HTTPS requests toward Keystone (if url is used in the authorization configuration)
+Add: support OAuth2 provider as authentication backend (#591)
+Add: support for authentication to NGSI subscription requests (#592)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: These of the new lines are already mentioned in the [release notes of version 2.7.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.7.0). I understand this PR (in fact, the old #600) covers only issue #592...

Maybe a merge conflict was incorrectly solved so these two lines as missappear here?

Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: In addition, with regards to:

```
Add: support for authentication to NGSI subscription requests
```

I undertand that NGSI refers to NGSIv1 **and** NGSIv2. Fine with it, just checking (given that parent issue #592 only mentions NGSIv1 in the title).

Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: Fixed with cdef73a623a9985bbb59d61b8ba144aa8fab48b2

Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: >I undertand that NGSI refers to NGSIv1 and NGSIv2. Fine with it, just checking (given that parent issue #592 only mentions NGSIv1 in the title).

You're right! :)

**Comment by dcalvoalonso** [comment]
Conflicts are fixed with f7e233f492dabcf76bad20ce637f217c00679c14

**Comment by fgalan** [comment]
It seems the merging of PR #763 causes a new conflict in CHANGES_NEXT_RELEASE. @dcalvoalonso could you solve so I can merge the PR?

**Comment by dcalvoalonso** [comment]
Conflicts are solved with https://github.com/telefonicaid/iotagent-node-lib/pull/764/commits/05a66e8dae55be5c96ae9ebebc0afb18c1e1ff67

**Pull request merged**